Packs/CiscoSMA/Integrations/CiscoSMA/CiscoSMA.py

```diff
@@ -0,0 +1,1947 @@
+            viewBy=view_by,
+            recipientAddresses=recipient_addresses,
+            senderAddresses=sender_addresses,
+            recipientList=recipient_list,
+            senderList=sender_list,
+        )
+
+        return self._http_request("POST", f"quarantine/{entry_type}", json_data=data)
+
+    def list_entry_edit_request(
+        self,
+        entry_type: str,
+        quarantine_type: str,
+        action: str,
+        view_by: str,
+        recipient_addresses: List[str] = None,
+        sender_list: List[str] = None,
+        sender_addresses: List[str] = None,
+        recipient_list: List[str] = None,
+    ) -> Dict[str, Any]:
+        """
+        Edit spam quarantine blocklist/safelist entries.
+
+        Args:
+            entry_type (str): Blocklist/Safelist list type.
+            quarantine_type (str): Quarantine type.
+            action (str): Edit action.
+            view_by (str): Edit list entry results by recipient/sender.
+            recipient_addresses (List[str], Optional): Recipient addresses list to edit Defaults to None.
+            sender_list (List[str], Optional): Sender addresses list to edit Defaults to None.
+            sender_addresses (List[str], Optional): Sender addresses list to edit Defaults to None.
+            recipient_list (List[str], Optional): Recipient addresses list to edit Defaults to None.
+
+        Returns:
+            Dict[str, Any]: API response from Cisco SMA.
+        """
+        data = assign_params(
+            action=action,
+            quarantineType=quarantine_type,
+            viewBy=view_by,
+            recipientAddresses=recipient_addresses,
+            senderAddresses=sender_addresses,
+            recipientList=recipient_list,
+            senderList=sender_list,
+        )
+
+        return self._http_request("POST", f"quarantine/{entry_type}", json_data=data)
+
+    def list_entry_delete_request(
+        self,
+        entry_type: str,
+        quarantine_type: str,
+        view_by: str,
+        recipient_list: List[str] = None,
+        sender_list: List[str] = None,
+    ) -> Dict[str, Any]:
+        """
+        Delete spam quarantine blocklist/safelist entries.
+
+        Args:
+            entry_type (str): Blocklist/Safelist list type.
+            quarantine_type (str): Quarantine type.
+            view_by (str): Delete list entry results by recipient/sender.
+            recipient_list (List[str], Optional): Recipient list to delete. Defaults to None.
+            sender_list (List[str], Optional): Sender list to delete. Defaults to None.
+
+        Returns:
+            Dict[str, Any]: API response from Cisco SMA.
+        """
+        data = assign_params(
+            quarantineType=quarantine_type,
+            recipientList=recipient_list,
+            senderList=sender_list,
+            viewBy=view_by,
+        )
+
+        return self._http_request("DELETE", f"quarantine/{entry_type}", json_data=data)
```

Comment:
what happens when an error is returned from the service? (non-200 code response)

Reply:
If the request was not successful it will raise an error and the user will get the error message from the API.